From 3cd536147a062155a83a3caa257a9d1022ce0d46 Mon Sep 17 00:00:00 2001
From: Cristian Sovaiala <cristian.sovaiala@freescale.com>
Date: Mon, 7 Jul 2014 14:22:56 +0300
Subject: [PATCH 755/987] dpaa_eth: Implement suspend/resume instead of
 suspend_noirq/resume_noirq

Due to external constraints (modules loading order and registry
configuration order in Auto Response mode) we need to implement
suspend/resume instead of suspend_noirq/resume_noirq callbacks.

Signed-off-by: Cristian Sovaiala <cristian.sovaiala@freescale.com>
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/net/ethernet/freescale/dpa/dpaa_eth.c        | 8 ++++----
 drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c  | 8 ++++----
 drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c | 8 ++++----
 drivers/net/ethernet/freescale/dpa/offline_port.c    | 8 ++++----
 4 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
index a387603..73bb635 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth.c
@@ -114,7 +114,7 @@ struct net_device *dpa_loop_netdevs[20];
 
 #ifdef CONFIG_PM
 
-static int dpaa_suspend_noirq(struct device *dev)
+static int dpaa_suspend(struct device *dev)
 {
 	struct net_device	*net_dev;
 	struct dpa_priv_s	*priv;
@@ -162,7 +162,7 @@ set_wol_failed:
 	return err;
 }
 
-static int dpaa_resume_noirq(struct device *dev)
+static int dpaa_resume(struct device *dev)
 {
 	struct net_device	*net_dev;
 	struct dpa_priv_s	*priv;
@@ -204,8 +204,8 @@ resume_failed:
 }
 
 static const struct dev_pm_ops dpaa_pm_ops = {
-	.suspend_noirq = dpaa_suspend_noirq,
-	.resume_noirq = dpaa_resume_noirq,
+	.suspend = dpaa_suspend,
+	.resume = dpaa_resume,
 };
 
 #define DPAA_PM_OPS (&dpaa_pm_ops)
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
index bfa069d..682e2b2 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_proxy.c
@@ -60,7 +60,7 @@ MODULE_PARM_DESC(debug, "Module/Driver verbosity level");
 static int __cold dpa_eth_proxy_remove(struct platform_device *of_dev);
 #ifdef CONFIG_PM
 
-static int proxy_suspend_noirq(struct device *dev)
+static int proxy_suspend(struct device *dev)
 {
 	struct proxy_device *proxy_dev = dev_get_drvdata(dev);
 	struct mac_device *mac_dev = proxy_dev->mac_dev;
@@ -78,7 +78,7 @@ port_suspend_failed:
 	return err;
 }
 
-static int proxy_resume_noirq(struct device *dev)
+static int proxy_resume(struct device *dev)
 {
 	struct proxy_device *proxy_dev = dev_get_drvdata(dev);
 	struct mac_device	*mac_dev = proxy_dev->mac_dev;
@@ -97,8 +97,8 @@ port_resume_failed:
 }
 
 static const struct dev_pm_ops proxy_pm_ops = {
-	.suspend_noirq = proxy_suspend_noirq,
-	.resume_noirq = proxy_resume_noirq,
+	.suspend = proxy_suspend,
+	.resume = proxy_resume,
 };
 
 #define PROXY_PM_OPS (&proxy_pm_ops)
diff --git a/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c b/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c
index 7b88592..17dc86f 100644
--- a/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c
+++ b/drivers/net/ethernet/freescale/dpa/dpaa_eth_shared.c
@@ -604,7 +604,7 @@ static int dpa_shared_netdev_init(struct device_node *dpa_node,
 
 #ifdef CONFIG_PM
 
-static int dpa_shared_suspend_noirq(struct device *dev)
+static int dpa_shared_suspend(struct device *dev)
 {
 	struct net_device	*net_dev;
 	struct dpa_priv_s	*priv;
@@ -629,7 +629,7 @@ port_suspend_failed:
 	return err;
 }
 
-static int dpa_shared_resume_noirq(struct device *dev)
+static int dpa_shared_resume(struct device *dev)
 {
 	struct net_device	*net_dev;
 	struct dpa_priv_s	*priv;
@@ -655,8 +655,8 @@ port_resume_failed:
 }
 
 static const struct dev_pm_ops shared_pm_ops = {
-	.suspend_noirq = dpa_shared_suspend_noirq,
-	.resume_noirq = dpa_shared_resume_noirq,
+	.suspend = dpa_shared_suspend,
+	.resume = dpa_shared_resume,
 };
 
 #define SHARED_PM_OPS (&shared_pm_ops)
diff --git a/drivers/net/ethernet/freescale/dpa/offline_port.c b/drivers/net/ethernet/freescale/dpa/offline_port.c
index 17ed9af..81d8dac 100644
--- a/drivers/net/ethernet/freescale/dpa/offline_port.c
+++ b/drivers/net/ethernet/freescale/dpa/offline_port.c
@@ -76,7 +76,7 @@ MODULE_DEVICE_TABLE(of, oh_port_match_table);
 
 #ifdef CONFIG_PM
 
-static int oh_suspend_noirq(struct device *dev)
+static int oh_suspend(struct device *dev)
 {
 	struct dpa_oh_config_s	*oh_config;
 
@@ -84,7 +84,7 @@ static int oh_suspend_noirq(struct device *dev)
 	return fm_port_suspend(oh_config->oh_port);
 }
 
-static int oh_resume_noirq(struct device *dev)
+static int oh_resume(struct device *dev)
 {
 	struct dpa_oh_config_s	*oh_config;
 
@@ -93,8 +93,8 @@ static int oh_resume_noirq(struct device *dev)
 }
 
 static const struct dev_pm_ops oh_pm_ops = {
-	.suspend_noirq = oh_suspend_noirq,
-	.resume_noirq = oh_resume_noirq,
+	.suspend = oh_suspend,
+	.resume = oh_resume,
 };
 
 #define OH_PM_OPS (&oh_pm_ops)
-- 
1.9.1

